cryptoki/src/session/object_management.rs

```diff
@@ -44,6 +44,13 @@ impl Session {
         while object_count > 0 {
```

Comment:
could it work change this instead to:
```rust
while object_count == MAX_OBJECT_COUNT {
```
?

Reply:
Not really, as it would exit the loop directly and not extend the resulting array with the last result from `C_FindObjects()`. It would result in a truncated, incomplete result.

I have changed the loop and removed one unsafe `C_FindObjects` section, see if you find it suitable. Works for me (tests are passing as expected).

Reply:
I think you were correct, actually. Please consider the new commit anyway, It should work as well and reduces code duplication.